src/main/java/com/dataguise/saas/service/ProtectionTaskService.java

```diff
@@ -0,0 +1,611 @@
+    }
+
+    /**
+     * Fetch all the mappings, filter on database name and connectionId.
+     */
+    private List<DgMaskerDbmsStructureMapping> getMapping(String databaseName, Integer connectionId) throws BadGatewayException {
+        List<DgMaskerDbmsStructureMapping> structureMappings = null;
+        try {
+            structureMappings = restDriver.getStructureMappingDetails();
+            if(connectionId!= null && connectionId!=0 &&
+                    !databaseName.isEmpty() && databaseName!= null
+                    && structureMappings!=null) {
+                structureMappings = structureMappings.stream().filter(mapping ->
+                                mapping.getConnectionId().equals(connectionId) &&
+                                mapping.getDbName().equalsIgnoreCase(databaseName)
+                                ).collect(Collectors.toList());
+            }
+        } catch (DgDrvException e) {
+            logger.error(e);
+            throw new BadGatewayException(e.getMessage() , HttpStatus.BAD_GATEWAY);
+        }
+        return structureMappings;
+    }
+
+    /**
+     * Create a structure with list of columns and their sensitive details.
+     */
+    private Integer createStructure(List<ProtectionColumnUIStruct> columnStructureList, String module) throws BadGatewayException  {
+        Integer structureId = 0;
+        DBMSStructUI structure = new DBMSStructUI();
+        String structureName = taskBuilderService.constructStructureName();
+        structure.setName(structureName);
+        structure.setDescription(structureName);
+        structure.setId(structureId);
+        structure.setModuleType(module);
+        for (ProtectionColumnUIStruct columnStructure : columnStructureList) {
+            DBMSStructColumnUI  dbmsColumnStruct = new DBMSStructColumnUI();
+            dbmsColumnStruct.setColumnName(columnStructure.getColumnName());
+            dbmsColumnStruct.setMaskingId(columnStructure.getRegexId());
+            dbmsColumnStruct.setTableName(columnStructure.getTableName());
+            structure.addColInfoListItem(dbmsColumnStruct);
+        }
+        try {
+
+            structureId = restDriver.createMaskingSructure(module,structure);
+        } catch (DgDrvException e) {
+            logger.error(e);
+           throw new BadGatewayException(ErrorConstants.ERROR_CREATING_STRUCTURE.getErrorMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return structureId;
+    }
+
+
+    /**
+     * Add db mappings to structure
+     */
+    private Boolean mapDatabaseToStructure(Integer structureId, Integer connectionId, String databaseName, String module) throws BadGatewayException {
+        Boolean mappingStatus;
+        try {
+            mappingStatus = restDriver.createStructureMapping(module, structureId, connectionId, databaseName);
+        } catch (DgDrvException e) {
+            logger.error("Error mapping structureId: "+ structureId + " with database: " + databaseName
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ProtectionTaskService.java L427
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a371500170>*